Documentation/design-docs/COM-activation.md

```diff
@@ -0,0 +1,107 @@
+# COM Activation for .NET Core on Windows
+
+## Purpose
+
+In order to more fully support the vast number of existing .NET Framework users in their transition to .NET Core, support of the COM Activation scenario in .NET Core is required. Without this support it is not possible for many .NET Framework consumers to even consider transitioning to .NET Core. The intent of this document is to describe aspects of the activation of a .NET class written for .NET Core within a COM environment. This support includes but is not limited to activation via the [`CoCreateInstance()`](https://docs.microsoft.com/en-us/windows/desktop/api/combaseapi/nf-combaseapi-cocreateinstance)API in C/C++ or from a [Windows Script Host](https://docs.microsoft.com/en-us/windows/desktop/com/using-com-objects-in-windows-script-host).
+
+This document is limited to the in-proc COM Activation scenario. The out-of-proc COM Activation scenario is deferred.
+
+### Requirements
+
+* Discover all installed versions of .NET Core.
+* Load the appropriate version of .NET Core for the class.
+* Instantiate an instance of the class and return a pointer to an [`IUnknown`](https://docs.microsoft.com/en-us/windows/desktop/api/unknwn/nn-unknwn-iunknown) that represents the newly constructed class.
+* Check if the activation host has already created a CLR instance of the correct version or create an appropriate instance.
+* Support the discrimination of concurrently loaded CLR versions.
+
+### Environment Matrix
+
+The following list represents an exhaustive activation matrix.
+
+|Server | Client | Current Support |
+| --- | --- | :---: |
+| COM* | .NET Core | Yes |
```

Comment:
Nope that is the next row. At present, .NET Core can activate a COM server (e.g. native DLL) via the registry _or_ when using `CoreRun` RegFree COM is also available. The .NET Server scenario (e.g. managed DLL) cannot be activated by a COM client (e.g. native EXE) at the moment because there is no way to instantiate a managed class without the CLR being loaded/started.

Reply:
Oh, I had read it to mean the scenario you're designing. Thanks!